protocol/x/stats/keeper/keeper.go

```diff
@@ -274,3 +280,75 @@ func (k Keeper) ExpireOldStats(ctx sdk.Context) {
 	metadata.TrailingEpoch += 1
 	k.SetStatsMetadata(ctx, metadata)
 }
+
+func (k Keeper) GetStakedAmount(ctx sdk.Context,
+	delegatorAddr string) big.Int {
+	startTime := time.Now()
+	var stakedAmount big.Int
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.CachedStakeAmountKeyPrefix))
+	bytes := store.Get([]byte(delegatorAddr))
+
+	// return cached value if it's not expired
+	if bytes != nil {
+		var cachedStakedAmount types.CachedStakeAmount
+		k.cdc.MustUnmarshal(bytes, &cachedStakedAmount)
+		if ctx.BlockTime().Unix()-int64(cachedStakedAmount.CachedAt) < int64(types.StakedAmountCacheDurationSeconds) {
```

Comment:
Let's change `CachedAt` to `int64` (same as `BlockTime().Unix()`) to avoid casting

Reply:
Nit: `<=` instead of `<`

Reply:
Could we add sanity check to avoid overflow:

`ctx.BlockTime().Unix() >= cachedStakedAmount.CachedAt >= 0`

If not true, panic since an invariant is broken